### PR TITLE
portis integration, add web3 state, remove old code

### PR DESCRIPTION
**Pull request by tamirse**

### CODE REVIEW

File: src/store/reducers/web3.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  web3: null,
+};
+
+const setWeb3 = (state, web3) => {
+  return { web3: web3 };
+};
+
+const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case actionTypes.SET_WEB3:
```
Comment (leonprou) [inline src/store/reducers/web3.js:13]: I don't like put complex objects into the redux state, prefer to define them in a service (see what I did with graphql client). But maybe people are doing this so guess it's ok

Comment (tamirse) [reply src/store/reducers/web3.js:13]: I don't know what people do, I just came up with the idea...
The thing is that the web3 is null at the beginning and only initialized with the provider the user choses with web3modal (metamask, portis etc)

What do?

File: src/components/Header/Header.js [hunk 2]
```diff
@@ -120,7 +123,12 @@ const Header = (props) => {
     if (!address) {
       onConnect();
     } else {
```
Comment (leonprou) [inline src/components/Header/Header.js:125]: when that else happens, can you explain in a comment

Comment (tamirse) [reply src/components/Header/Header.js:125]: it happens when the user is already connected and clicks on the wallet button. added comment in code.

**Comment by tamirse** [comment]
@leonprou I exported the web3modal to a hook, similar to the fuse example you showed me

**Comment by leonprou** [comment]
@tamirse I see, looks good

**Pull request merged**